Automattic/wp-calypso
# Include `user_is_dev_account` property in page view tracking

Related to https://github.com/Automattic/dotcom-forge/issues/5444

## Proposed Changes

To understand how our developer users interact with WordPress.com, we are introducing a new property `user_is_dev_account` to all `calypso_page_view` events tracked using `<PageEventTracker>`.

The value is set to `1` if the user has the `is_dev_account` ("I am a developer") setting enabled, `0` otherwise.

 **Note:**
   * On initial page load, there's a chance that `user_is_dev_account` won't be tracked, since the page view event might be fired _before_ user settings are fetched. Subsequent page views should be fine, since user settings should then be available in the state. We have a similar behaviour for [`site_type`](https://github.com/Automattic/wp-calypso/blob/35b9c1fc0156926d4383c180bbcba851d203e0d2/client/state/analytics/actions/enhance-with-site-type.js#L13-L28). 
   * Some page views are not tracked using `<PageViewTracker>`, in which case `user_is_dev_account` will not be included. If there are specific pages that we are interested in, we can enhance those tracking calls in followup PRs.

~A new function `recordEnhancedTrackEvents()` is introduced allowing us to include additional properties to any track events using enhancers. This allows us to include the `user_is_dev_account` property for the `calypso_me_developer_learn_more` track event, so we can differentiate clicks by developers and non-devs.~ EDIT: There were a bunch of issues we ran into with redux for click tracking. We're going to descope this PR so that we can move forward with rolling out the page view event updates and followup with a separate PR to add `user_is_dev_account` to the `calypso_me_developer_learn_more` click tracking event.

## Testing Instructions



* Navigate to various Calypso pages and assert that you see the `user_is_dev_account` property in the `calypso_page_view` event and that it's set to the correct value. (Keep in mind the exceptions noted in the above description.)

<img width="690" alt="Screen Shot 2024-01-31 at 9 58 55 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/b184affd-0192-4474-8eeb-a30be005bfb1">

* Navigate to `/me/developers`, click on any "Learn more" link. Assert that `user_is_dev_account` property exists in the `calypso_me_developer_learn_more` event and that it's set to the correct value.

<img width="421" alt="Screen Shot 2024-01-31 at 10 06 41 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/e07322c5-6a49-401d-b2e3-3bb09b93af94">

* Update your "I am a developer" setting and re-test. Assert that the `user_is_dev_account` property reports the correct value.

## Pre-merge Checklist
